app/scripts/background.js

```diff
@@ -251,17 +251,10 @@ function setupController (initState, initLangCode) {
     getIpfsGateway: controller.preferencesController.getIpfsGateway.bind(controller.preferencesController),
     provider: controller.provider,
   })
-
-  // setup state persistence
-  pump(
-    asStream(controller.store),
-    debounce(1000),
-    storeTransform(versionifyData),
-    createStreamSink(persistData),
-    (error) => {
-      log.error('MetaMask - Persistence pipeline failed', error)
-    },
-  )
+  onControllerInit(controller)
+  controller.on('controllerInitialized', () => {
+    onControllerInit(controller)
```

Comment:
I I don't understand why `onControllerInit` can be called up to twice if `controllerInitialized` is fired. Is that intentional?